### DISCUSSION TITLE
I am compiling kratos , and meet a question when run the configure.bat

### DISCUSSION THREAD

**Discussion by Ada2045lalala** [KratosMultiphysics/Kratos#11934] [Q&A] [answered]
### i use Windows Powershell to run the configure.bat：
`@echo off
rem Please do not modify this script

rem For any question please contact with us in:
rem  - https://github.com/KratosMultiphysics/Kratos

rem Optional parameters:
rem You can find a list will all the compiation options in INSTALL.md or here:
rem  - https://github.com/KratosMultiphysics/Kratos/wiki/Compilation-options

rem Set compiler
set CC=cl.exe
set CXX=cl.exe

rem Set variables
if not defined KRATOS_SOURCE set KRATOS_SOURCE=%~dp0..
if not defined KRATOS_BUILD set KRATOS_BUILD=%KRATOS_SOURCE%/build

rem Warning: In windows this option only works if you run through a terminal with admin privileges
rem set KRATOS_INSTALL_PYTHON_USING_LINKS=ON

rem Set basic configuration
if not defined KRATOS_BUILD_TYPE set KRATOS_BUILD_TYPE=Release
if not defined BOOST_ROOT set BOOST_ROOT=D:\KratosMultiPhysics\boost_1_84_0
if not defined PYTHON_EXECUTABLE set PYTHON_EXECUTABLE=C:\Windows\py.exe

rem Set applications to compile
set KRATOS_APP_DIR=applications
set KRATOS_APPLICATIONS=
CALL :add_app %KRATOS_APP_DIR%\LinearSolversApplication;
CALL :add_app %KRATOS_APP_DIR%\StructuralMechanicsApplication;
CALL :add_app %KRATOS_APP_DIR%\FluidDynamicsApplication;

rem Clean
del /F /Q "%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%\cmake_install.cmake"
del /F /Q "%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%\CMakeCache.txt"
del /F /Q "%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%\CMakeFiles"

rem Enable this if your build is slow and you have a multi-core machine
rem set KRATOS_PARALLEL_BUILD_FLAG=/MP4

rem Configure
@echo on
cmake -G"Visual Studio 16 2019" -H"%KRATOS_SOURCE%" -B"%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%"          ^
-DUSE_EIGEN_MKL=OFF                                                                                 ^
-DCMAKE_CXX_FLAGS=" %KRATOS_PARALLEL_BUILD_FLAG% "                                                  ^
-DKRATOS_GENERATE_PYTHON_STUBS=ON

rem Build
cmake --build "%KRATOS_BUILD%/%KRATOS_BUILD_TYPE%" --target install -- /property:configuration=%KRATOS_BUILD_TYPE% /p:Platform=x64
goto:eof

rem Function to add apps
:add_app
set KRATOS_APPLICATIONS=%KRATOS_APPLICATIONS%%1;
goto:eof
`
### and got this result:
![2024-01-09 (5)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/4ede0019-1be0-4611-97dd-05c481ea9433)

![2024-01-09 (2)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/46e4d0db-0c76-4ca2-88a1-d5850245db4d)

![2024-01-09 (3)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/12b30d26-ddd5-4045-bf7a-15bc8f48f6d4)
how can i solve it?

**Comment by roigcarlo** [comment]
Hi @Ada2045lalala. There seems to be two problems in the log you send.

- 1) Please remove / uninstall or hide MinGW from the configure. This is what is causing the C2144 errors, as mingw libraries are incompatible with visual studio ones. 
- 2) Please ensure you have Microsoft Visual Studio updated. MSVC version should be 16 to 19, but it appears to 14 in your log.

**Reply by Ada2045lalala** [reply]
Thanks for your help! @roigcarlo 

i. I uninstalled MinGW from my PC and the C2144 errors disappeared. But there is so many C4819 warnings when I run the configure.bat . Will the C4819 warnings cause any troubles?
![2024-01-09 (7)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/c7781000-fd30-42b7-b232-34a2b9dc951e)
![2024-01-09 (10)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/715bd0f3-ef1d-4278-bad4-774ba80cc2cd)

ii.as for MSVC version, I run a code, and get the result:
![2024-01-09 (9)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/cc144ce1-9db1-4aa8-a50d-b5e0270256e8)

**Reply by roigcarlo** [reply, marked as answer]
Hi @Ada2045lalala 

C4819 its fine, its most likely a warning because our EOF character uses ANSI source files in windows (this is a encoding format for text files) and probably gets messy when trying to convert to a Chinese character table, but should not affect the compilation result.

I am not sure how to get rid of it but I will take a look and post it here in case I find something.

MSVC 1929 is fine as well :+1:

**Reply by Ada2045lalala** [reply]
Thanks！
another question. (I don't major in computer science or similar major, and I am sorry if I am asking a low-level question)
I don't know how to implement this sentence" set them permanently using the Edit the system environment variables option in the Control panel".
![2024-01-09 (11)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/54d6d396-4635-418b-902d-77f4728b7b78)

**Reply by roigcarlo** [reply]
Hi @Ada2045lalala

you can open it from the start menu:

![answer1](https://github.com/KratosMultiphysics/Kratos/assets/1935791/9ab43a08-189d-4a43-a53b-b7f9b9bb5493)

then select the Environmental variables button (in the bottom)

![answer2](https://github.com/KratosMultiphysics/Kratos/assets/1935791/3bfab4dc-9d2d-49c5-a8e5-9bd504642b44)

And you will see this window:

![answer3](https://github.com/KratosMultiphysics/Kratos/assets/1935791/e3b82ce6-ceb0-4f18-9f5a-e85e8a63598f)

There you can add/remove/change any system variable that you want.

**Reply by Ada2045lalala** [reply]
thanks again🌹
I guess I've finished the compilation.
![2024-01-10](https://github.com/KratosMultiphysics/Kratos/assets/155991150/3f2ce74d-b851-4062-afe9-29f4a6fddb19)

![2024-01-10 (1)](https://github.com/KratosMultiphysics/Kratos/assets/155991150/93872fb2-685c-4be0-b4d3-6d102853d80c)

**Reply by roigcarlo** [reply]
Yep everything seems to work fine!

If you have any other question don't doubt to ask.
